diff -up a/dwm.c b/dwm.c
--- a/dwm.c	2020-07-09 16:49:10.023585649 +0200
+++ b/dwm.c	2020-07-09 16:49:43.497542191 +0200
@@ -910,7 +910,7 @@
        return;	
 
 	/* draw status first so it can be overdrawn by tags later */
-	if (m == selmon) { /* status is only drawn on selected monitor */
+	if (m == selmon || 1) { /* status is only drawn on selected monitor */
        tw = m->ww - drawstatusbar(m, bh, stext);
    }

@@ -1987,9 +1987,11 @@ updatesizehints(Client *c)
 void
 updatestatus(void)
 {
+	Monitor* m;
 	if (!gettextprop(root, XA_WM_NAME, stext, sizeof(stext)))
 		strcpy(stext, "dwm-"VERSION);
-	drawbar(selmon);
+	for(m = mons; m; m = m->next)
+		drawbar(m);
 }
 
 void
